adapters/handlers/rest/authz/handlers_authz.go

```diff
@@ -110,30 +110,35 @@ func (h *authZHandlers) createRole(params authz.CreateRoleParams, principal *mod
 }
 
 func (h *authZHandlers) addPermissions(params authz.AddPermissionsParams, principal *models.Principal) middleware.Responder {
-	if *params.Body.Name == "" {
-		return authz.NewAddPermissionsBadRequest().WithPayload(cerrors.ErrPayloadFromSingleErr(errors.New("role name is required")))
-	}
-
-	if slices.Contains(authorization.BuiltInRoles, *params.Body.Name) {
-		return authz.NewAddPermissionsBadRequest().WithPayload(cerrors.ErrPayloadFromSingleErr(fmt.Errorf("you can not update builtin role %s", *params.Body.Name)))
+	if slices.Contains(authorization.BuiltInRoles, params.ID) {
+		return authz.NewAddPermissionsBadRequest().WithPayload(cerrors.ErrPayloadFromSingleErr(fmt.Errorf("you can not update builtin role %s", params.ID)))
 	}
 
 	if len(params.Body.Permissions) == 0 {
 		return authz.NewAddPermissionsBadRequest().WithPayload(cerrors.ErrPayloadFromSingleErr(fmt.Errorf("role has to have at least 1 permission")))
 	}
 
 	policies, err := conv.RolesToPolicies(&models.Role{
-		Name:        params.Body.Name,
+		Name:        &params.ID,
 		Permissions: params.Body.Permissions,
 	})
 	if err != nil {
 		return authz.NewAddPermissionsBadRequest().WithPayload(cerrors.ErrPayloadFromSingleErr(fmt.Errorf("invalid permission %s", err.Error())))
 	}
 
-	if err := h.authorizer.Authorize(principal, authorization.UPDATE, authorization.Roles(*params.Body.Name)...); err != nil {
+	if err := h.authorizer.Authorize(principal, authorization.UPDATE, authorization.Roles(params.ID)...); err != nil {
 		return authz.NewAddPermissionsForbidden().WithPayload(cerrors.ErrPayloadFromSingleErr(err))
 	}
 
+	roles, err := h.controller.GetRoles(params.ID)
```

Comment:
**suggestion (if-minor):** to make sure it's consistent, wdyt if we use HasPermissions() , to make sure it's not affected by any EC

Reply:
I think making these methods consistent is a good idea but `HasPermissions` isn't the correct method to use here. In this method, we check for pre-existence of the role by its name and not whether it has any permissions

Reply:
ok we can leave it as it's but can we add `todo-rbac: make sure it's consistent when we have getConsistentRole() `